mantidproject/mantid
# Introduce MDFrame and MDUnits

Fixes #12185

**Tester**
Please code review these changes and check that the system tests and unit tests are passing. That should catch most problems.

Then run this:

``` python
Load(Filename='SXD23767.raw',OutputWorkspace='SXD23767',LoadMonitors='Exclude')

QLab = ConvertToDiffractionMDWorkspace(InputWorkspace='SXD23767', OutputDimensions='Q (lab frame)',
                                       SplitThreshold=50, LorentzCorrection='1',
                                       MaxRecursionDepth='13',Extents='-15,15,-15,15,-15,15')

for i in range(QLab.getNumDims()):
    frame = QLab.getDimension(i).getMDFrame()
    print frame.name(), frame.getUnitLabel()
```
- Check that you get frame and unit information provided. Lastly, 
- Save the QLab file and reload it via `SaveMD`. Then run the query over the dimensions at the end. You should find that that information has been persisted.
